### DISCUSSION TITLE
[Kavita] I think the bookmarks is weird.

### DISCUSSION THREAD

**Discussion by imjinlok** [Kareadita/Kavita#3072] [Ideas]
### Idea Description

I think the bookmarks function is just a gallery now.
It just take pictures of the pages and put them in an album called Bookmarks.
This is not the bookmark of common sense I think.

Let's think of a real book.

The original purpose of the bookmark is to put it between pages and continue to read it later.
But the feature I are focusing on is that I want to mark favorite pages and see them again later.
It's like folding the corners of your favorite pages.
Then, later, I pick up the book from the bookcase and open the folded page, I can read it again where I like.

But now the bookmarks are gathered separately in the bookmarks library.
Now, when I pick up a book from a library (not a bookmark library) I can not quickly access the bookmarked point as if I were opening a folded page.
Now, I can not continue to read it unless I bookmark all my favorite pages.

This bookmark feature is very weird to my intuition, and useless to me.
At least, I want you to add feature that can jump to the previous, next bookmark in comic viewer.

### Idea Category

User Experience

### Duration of Using Kavita

1 month

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
If I'm understanding you correctly, you are wanting the feature to be more where you can resume from the page you previously book marked in the chapter as normal?

**Reply by imjinlok** [reply]
I give you an example because I'm not good at English.

When I read the book 'A', Kavita shows the comic book through the image viewer
Image viewer has some function buttons and is assigned keyboard shortcuts (G, B, ESC, SPACE...)
I want two buttons added to it. Go to the previous bookmark, go to the next bookmark.

In Book 'A', I already made bookmarks on page 10, 50, and 70.
And now I'm looking at page 30.
If I click 'Go to the previous bookmark' button, it jump to page 10.
If I click 'Go to the next bookmark' button. it jump to page 50.
If i click 'Go to the next bookmark' once again, jump to page 70.

**Comment by scare376** [comment]
This feels like a duplicate of #2954

**Reply by imjinlok** [reply]
I've already read that.
He wants to jump from the bookmark library to the original.

I think a little differently.
I don't want to go to the bookmark library just to jump to the bookmark point when I reading a book 'A'.
I should be able to jump to the bookmark while i was reading 'A'.
That is the folding of the coners of the page i said.

**Comment by majora2007** [comment]
Agreed. @imjinlok please move your comment and upvote there

**Reply by imjinlok** [reply]
I'm sorry. I'm here to make a suggestion and I don't know how to use GitHub.
I can't find the delete button or the moving button.

**Comment by majora2007** [comment]
Oh okay. We used to have that actually but I removed it when I created the dedicated bookmark page.

**Comment by majora2007** [comment]
Recaping for myself. User is asking for the ability to list the bookmarks from within the original chapter reader and jump to different bookmarks from WITHIN the reader.

**Reply by VincentOUALLET** [reply]
Hy, thank you for the developpment of this app, really enjoy & appreciate it!
Will this feature be implemented someday?
Thanks

**Reply by majora2007** [reply]
Maybe, the foundation was updated to enable, but it's not a priority for me.

**Reply by VincentOUALLET** [reply]
Yes I understand that. My only priority on that point is more to have displayed the current tome and page (the user can at least go back to the place the captured bookmark had been taken), I think it's weird to have that that way without any information of were it come from. But I understand it's complicated to have something that works with all type of data, and Im not familiar with everything you did, I even didnt know we could display img from archive in browzers before discovering Kavita. I'm gonna do it on my instance when I'll have time cause it's great on my side to just see again my favorite moments in books I like, pretty sure a lot of comics and mangas fan think the same (so much beautifull pictures). Hope I can do it properly and maybe give you this part of the work someday. What do you call "the foundation"? People from Kavita?

**Reply by majora2007** [reply]
"the foundation" refers to the underlying code to enable the features to jump back to the original spot for the bookmark and the like. 

For me, I just wanted a copy of images I liked from manga, as I came from Ubooquity and that was all it offered. Your idea to return to the reader is not wrong, just not a priority (but I did already code up the backend to support).
